drivers/ephy.c

```diff
@@ -244,8 +422,52 @@ static char *ephy_parsePinArg(char *cfg, const char *pfx, size_t pfx_len, gpio_i
 }
 
 
-/* printf("This is an Ethernet PHY driver. use: %s id mdio-bus phy-addr [irq=[-]n,/dev/gpio/X] [reset=[-]n,/dev/gpio/X]\n", argv[0]); */
-/* ARGS: [bus.]id[:reset:[-]n:/dev/gpioX][:irq:[-]n:/dev/gpioX] */
+static __attribute__((unused)) char *ephy_parsePhyModel(eth_phy_state_t *phy, char *cfg)
+{
+	char *p = cfg;
+	if (strncmp(p, "ksz8081rn", 9) == 0) {
+		if (p[9] == 'a') {
+			phy->model = ephy_ksz8081rna;
+		}
+		else if (p[9] == 'b') {
+			phy->model = ephy_ksz8081rnb;
+		}
+		else if (p[9] == 'd') {
+			phy->model = ephy_ksz8081rnd;
+		}
+		else {
+			printf("lwip: ephy: unsupported PHY model: `ksz8081rn%c`\n", p[9]);
+			return NULL;
+		}
+	}
+	else if (strncmp(p, "rtl8201fi", 9) == 0) {
+		phy->model = ephy_rtl8201fi;
+	}
+	else if (strncmp(p, "rtl8211fdi", 10) == 0) {
+		phy->model = ephy_rtl8211fdi;
+	}
+	else {
+		printf("lwip: ephy: unsupported PHY model\n");
+		return NULL;
+	}
+
+	p = strchr(cfg, ':');
```

Comment:
A string like `rtl8201fiaaaaaaa` would still be accepted as a valid match for `ephy_rtl8201fi`. Additionally, `p` is not advanced after a successful `strcmp`.